assets/basic/shaders/PushConstantsTexture.hlsl

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//
+// Vulkan Note:
+//   Must comply with Vulkan's packing rules as specified 
+//   by the compiler.
+//
+struct DrawParams
+{
+    float4x4 MVP;
+    uint     TextureIndex;
+};
+
+#if defined(__spirv__)
+[[vk::push_constant]]
+#endif
+ConstantBuffer<DrawParams> Draw : register(b0);
+
+Texture2D    Textures[3]: register(t1);
+SamplerState Sampler0   : register(s4);
+
+struct VSOutput {
+	float4 Position : SV_POSITION;
```

Comment:
nit: there is mixed use of tabs and spaces in the file, would be nice to be consistent, but up to you

Reply:
Not sure how this happened. Fixed to all spaces.